src/Utilities/CssDocument.php

```diff
@@ -110,175 +113,62 @@ public function getStyleRulesData(array $allowedMediaTypes): array
-     * - key "media" will contain "@media " followed by the media query list, for all allowed media queries,
-     *   or an empty string for CSS not within a media query
-     *
-     * Example:
-     *
-     * The CSS code
-     *
-     *   "@import "file.css"; h1 { color:red; } @media { h1 {}} @media tv { h1 {}}"
-     *
-     * will be parsed into the following array:
-     *
-     *   0 => [
-     *     "css" => "h1 { color:red; }",
-     *     "media" => ""
-     *   ],
-     *   1 => [
-     *     "css" => " h1 {}",
-     *     "media" => "@media "
-     *   ]
-     *
-     * @param array<array-key, string> $allowedMediaTypes
-     *
-     * @return array<int, array{css: string, media: string}>
-     */
-    private function splitCssAndMediaQuery(array $allowedMediaTypes): array
-    {
-        $mediaTypesExpression = '';
-        if (!empty($allowedMediaTypes)) {
-            $escapedAllowedMediaTypes = \array_map(
-                static function (string $mediaType): string {
-                    return \preg_quote($mediaType, '#');
-                },
-                $allowedMediaTypes
-            );
-            $mediaTypesExpression = '|' . \implode('|', $escapedAllowedMediaTypes);
+        if ($rule instanceof CssImport) {
+            return $this->isImportRuleAllowed;
         }
 
-        $cssSplitForAllowedMediaTypes = \preg_split(
-            '#(@media\\s++(?:only\\s++)?+(?:(?=[{(])' . $mediaTypesExpression . ')' . self::MEDIA_RULE_BODY_MATCHER
-            . ')#misU',
-            $this->styleRules,
-            -1,
-            PREG_SPLIT_DELIM_CAPTURE
-        );
+        $this->isImportRuleAllowed = false;
 
-        // filter the CSS outside/between allowed @media rules
-        $cssCleaningMatchers = [
-            'import/charset directives' => '/\\s*+@(?:import|charset)\\s[^;]++;/i',
-            'remaining media enclosures' => '/\\s*+@media\\s' . self::MEDIA_RULE_BODY_MATCHER . '/isU',
-        ];
+        if (!$rule instanceof CssAtRule) {
+            return false;
+        }
 
-        $splitCss = [];
-        foreach ($cssSplitForAllowedMediaTypes as $index => $cssPart) {
-            $isMediaRule = $index % 2 !== 0;
-            if ($isMediaRule) {
-                \preg_match('/^([^{]*+){(.*)}[^}]*+$/s', $cssPart, $matches);
-                $splitCss[] = ['css' => $matches[2], 'media' => $matches[1]];
-            } else {
-                $cleanedCss = \trim(\preg_replace($cssCleaningMatchers, '', $cssPart));
-                if ($cleanedCss !== '') {
-                    $splitCss[] = ['css' => $cleanedCss, 'media' => ''];
-                }
-            }
+        switch ($rule->atRuleName()) {
+            case 'media':
+                return false;
+            case 'font-face':
+                return $rule instanceof CssRuleSet
+                    && $rule->getRules('font-family') !== []
+                    && $rule->getRules('src') !== [];
+            default:
+                return true;
```

Comment:
Please let's have early returns only for guard clauses.

Reply:
I think there can be a case for breaking this rule, if all of the following apply:
- The `swtich` statement is the only statement in the method;
- All `case` branches have a `return` statment and not a `break` statement, except where a 'follow-through' is allowed or for the last `case` or `default` if `null` or `void` is the obviosuly desired return value in that case.

However, maintaining that standard through future code changes may be harder than it seems.  So I will probably change it when I revisit the code changes on this - perhaps tomorrow.

Reply:
> if all of the following apply

Such a method would be atomic.  If there was any future need to add pre-amble or post-amble code, then it should be moved to a new atomic method, with the original method becoming a wrapper with the additional pre- or post- code.  Though we know from experience it often doesn't turn out that way.

Reply:
> 
> 
> Please let's have early returns only for guard clauses.

I've changed the `switch` statement, though not the preceding code.

I don't think I agree with a "one size fits all" approach regarding where a return statement is or isn't acceptable, provided the complexity of an individual method can be kept low, as I've indicated above.